destination.go

```diff
@@ -51,38 +50,11 @@ type Destination interface {
 	// this function.
 	Open(context.Context) error
 
-	// WriteAsync receives a Record and can cache it internally and write it at
-	// a later time. Once the record is successfully written to the destination
-	// the plugin must call the provided AckFunc function with a nil error. If
-	// the plugin failed to write the record to the destination it must call the
-	// supplied AckFunc with a non-nil error. If any AckFunc is left uncalled
-	// the connector will not be able to exit gracefully. The WriteAsync
-	// function should only return an error in case a critical error happened,
-	// it is expected that all future calls to WriteAsync will fail with the
-	// same error.
-	// If the plugin caches records before writing them to the destination it
-	// needs to store the AckFunc as well and call it once the record is
-	// written.
-	// AckFunc will panic if it's called more than once.
-	// If WriteAsync returns ErrUnimplemented the SDK will fall back and call
-	// Write instead. A connector must implement exactly one of these functions,
-	// not both.
-	WriteAsync(context.Context, Record, AckFunc) error
-
-	// Write receives a Record and is supposed to write the record to the
-	// destination right away. If the function returns nil the record is assumed
-	// to have reached the destination.
-	// WriteAsync takes precedence and will be tried first to write a record. If
-	// WriteAsync is not implemented the SDK will fall back and use Write
-	// instead. A connector must implement exactly one of these functions, not
-	// both.
-	Write(context.Context, Record) error
-
-	// Flush signals the plugin it should flush any cached records and call all
-	// outstanding AckFunc functions. This function needs to be implemented only
-	// if the struct implements WriteAsync. No more calls to WriteAsync
-	// will be issued after Flush is called.
-	Flush(context.Context) error
+	// Write writes len(r) records from r to the destination right away without
+	// caching. It should return the number of records written from r
+	// (0 <= n <= len(r)) and any error encountered that caused the write to
+	// stop early. Write must return a non-nil error if it returns n < len(r).
+	Write(ctx context.Context, r []Record) (n int, err error)
```

Comment:
It would be great if we could test it in an acceptance test, although the hard part is provoking an error in the destination, we have no idea what the destination is and how to make it fail. I don't see an easy way of doing that right now 😕